dali/kernels/slice/slice_hwc2chw_normalize_gpu.cu

```diff
@@ -0,0 +1,532 @@
+// Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <tuple>
+#include <type_traits>
+#include "dali/core/backend_tags.h"
+#include "dali/core/convert.h"
+#include "dali/core/error_handling.h"
+#include "dali/core/permute.h"
+#include "dali/core/static_switch.h"
+#include "dali/kernels/slice/slice_hwc2chw_normalize_gpu.h"
+
+namespace dali {
+namespace kernels {
+
+namespace slice_flip_normalize {
+
+template <typename Out, typename In>
+struct Hwc2ChwSampleDesc {
+  Out *__restrict__ out;
+  const In *__restrict__ in;
+  const float *__restrict__ norm_add;
+  const float *__restrict__ norm_mul;
+  const void *__restrict__ fill_values;
+
+  uint32_t first_block;
+  int64_t sample_size;
```

Comment:
Swap for better structure packing (or better, pass an array of first_block separately, see below).

Reply:
Done